hip-rocclr/amdgpu-targets.patch

```diff
@@ -0,0 +1,19 @@
+*** HIP-rocm-3.7.0/hip-config.cmake.in.bak	2020-09-09 15:37:55.780535875 +0200
+--- HIP-rocm-3.7.0/hip-config.cmake.in	2020-09-09 15:38:13.160302239 +0200
+***************
+*** 79,85 ****
+            ${HIP_CLANG_INCLUDE_SEARCH_PATHS}
+        NO_DEFAULT_PATH)
+    find_dependency(AMDDeviceLibs)
+!   set(AMDGPU_TARGETS "gfx900;gfx906;gfx908" CACHE STRING "AMD GPU targets to compile for")
```

Comment:
I'm wondering... does every roc library have up to date defaults for `AMDGPU_TARGETS`? If you rename that here, another library might break.

Reply:
That's a good point. I don't know. Best way to find this out is to rebuild all packages.

Reply:
More precisely: All packages that depend on HIP.

Reply:
I checked all packages in `rocm-libs`. They all include up to date defaults, that is `gfx803;gfx900;gfx906;gfx908`.

Reply:
I was wondering it would make sense to simply define `AMDGPU_TARGETS "gfx803;gfx900;gfx906;gfx908;gfx1011"` and (hopefully) also get Navy support.

Reply:
That would (hopefully) override the default in all the other packages so that nothing would need to be changed there.

Reply:
No, unfortunately not. As I wrote earlier:
> > Do you think it would make sense to include Navi
> 
> I've added it to the list but then building `rocrand` fails because of unsupported asm instructions (as expected). That's why I've removed it again.

Reply:
I

> No, unfortunately not. As I wrote earlier:
> 
> > > Do you think it would make sense to include Navi
> > 
> > 
> > I've added it to the list but then building `rocrand` fails because of unsupported asm instructions (as expected). That's why I've removed it again.

Ah, sorry, I missed that one.

It would seem that `miopen-hip` is also not built for gfx803. If I read the CMake code correctly, they use the global default (if they find `hip-config.cmake` in /opt/rocm/*)

Reply:
Oh wow, I totally misread this patch, yesterday. I feel a little stupid :D I thought you were removing `AMDGPU_TARGETS` entirely :facepalm: